### DISCUSSION TITLE
Author submissions - what type of maintenance committment do we expect from our maintainers?

### DISCUSSION THREAD

**Discussion by lwasser** [pyOpenSci/handbook#20] [Peer Review]
Currently we accept all packages that are within our review scope. However there are a few scenarios where we need to have processes in place to ensure we are accepting packages that have some sort of maintenance plan.

In one instance, we had a package submitted by a student working in a lab. while this type of submission is wonderful, now I am trying to get in touch with maintainers and that individual who submitted is no longer with the lab . A graduate student may spend 2-5 years ish in that sort of position. but a gra position supporting work on a tool may only last a semester. so while we want to encourage student submissions where they make sense... we need to specify DURATION of time available to maintain the package. 

One of the goals for pyOpenSci is to create an ecosystem of curated, vetted packages. This is a specific spot where we differ from organizations like JOSS. 

So we need to think about language around what we expect from submissions.
* Who is the corresponding author on the submission, and how long are they available to work on the package
* what is the package maintenance plan (minimum of two years).
I think all authors should get credit for their work maintaining a package but this scenario is difficult because we need to be able to reach maintainers and have some sort of initial commitment of maintaining the package.

**Comment by SimonMolinsky** [comment]
Maybe the good idea is to provide a badge system:

- green badge - a package is maintained (a specific number of PR's per 6-month period?)
- yellow badge - the last update was between 6mo - 1yr,
- red badge - the last update was done > 1yr ago.

This approach doesn't exclude newcomers, and it clearly shows the community health of a package. Based on badges, packages in the pyOpenSci DB may be grouped and presented to a wider audience. If the maintainer will not care, then the package will lose its category (badge, pyOpenSci support).

Two weeks ago I had a short lecture about data viz ecosystem in Python & Java Script, and I didn't bother to mention libraries that were not updated in the last year.
